Shopify/dawn
# [B2B] Restrict pricing

### What approach did you take?

I used the following conditional to hide pricing and buying functions:

```liquid
{%- if shop_restricted_b2b_buying_enabled == false or shop_restricted_b2b_buying_enabled and customer.b2b? -%}
  …
{%- endif -%}
```

This conditional checks if `shop_restricted_b2b_buying_enabled` is `false` or if `shop_restricted_b2b_buying_enabled` is `true` and the customer is a B2B customer.

I added a temporary assignment variable (`shop_restricted_b2b_buying_enabled`) to simulate the behaviour of the upcoming API setting in order to prototype conditional logic for displaying prices and buying functions for B2B customers (this will be removed later):

```liquid
{%- assign shop_restricted_b2b_buying_enabled = true %}
```

Assuming the new `shop_restricted_b2b_buying_enabled` setting won't even return `nil` and just `true` or `false`, I'm omitting the check against `!= blank`.

### Other considerations

For the product cards, using the conditional above, I wrapped the price element in the `price.liquid` snippet. Instead, we may decide to add this conditional to each and every section but for now, I just added it to the `price.liquid` snippet for the simplicity of this prototype.

I hid the prices from the price meta property and schema price on the product page and featured product section. This is normally revealed in the DOM ([screenshot 1](https://screenshot.click/07-08-0y73y-voabt-23-86c6f.png), [screenshot 2](https://screenshot.click/07-09-xok5q-utrbo-23-7c0vh.png)). However, prices are still shown in the DOM (Chome dev tools > search for `web-pixels-manager-setup` and `ShopifyAnalytics`: [screenshot 1](https://screenshot.click/07-22-hbgns-7pyz1-23-1kg7i.png), [screenshot 2](https://screenshot.click/07-21-acc4b-hz3ys-23-zs3n9.png)) but this isn't controlled in the theme 🤔

### Testing steps/scenarios

Hide whitespace when reviewing code.

Places where pricing should be hidden for non-B2B customers:

- Product cards
  - Collection templates
  - Main search results
  - Featured Collection section
  - Collage section
  - Complementary products block
  - Related/recommended products section
  - Predictive search product results ([setting screenshot](https://screenshot.click/06-13-8gsr2-yresw-23-smqcw.png))
  - Sale and Sold out badges
  - Cart notification? Could buyers try to show such a component and get insights of a price?
  - Pricing filters ⚠️ _pending_
- Meta property for price
- Product page templates (Price block)
  - Product schema price in the DOM (dev tools)
- Featured product section
  - Product schema price in the DOM (dev tools)

Buying functions that shouldn't be present for non-B2B customers:

- Quick add buttons on Product cards
  - Collection templates
  - Featured Collection section
  - Complementary products block
- Cart icon in header: ⚠️ _icon spacing pending_
- Buy Buttons on PDP and (Featured Product section)
- Quantity Pickers on PDP (and Featured Product section)
- Quick Order List on PDP (entire section): ⚠️ _pending_

### Demo links

- [Editor](https://admin.shopify.com/store/os2-demo/themes/162642788374/editor)
